### PR TITLE
Test and support Python 3.6

### PR DESCRIPTION
**Pull request by bz2**
Add py36 to the Travis CI config, and `setup.py` classifiers.

Also tweak local tox config, install twisted for 3.6 but make `noptionals` variant use 2.7 and 3.5 only.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = coverage-clean,py26,py27,py34,py35,py36,pypy,{py27,py35,py36}-nooptionals,coverage-report
+envlist = coverage-clean,py26,py27,py34,py35,py36,pypy,{py27,py35}-nooptionals,coverage-report
```
Comment (brian-brazil) [inline tox.ini]: Remove 3.5 from the nooptionals, we only need a 2 and a 3 there.

Comment (brian-brazil) [inline tox.ini]: Let's test the noooptionals with 3.6 rather than 3.5

Comment (bz2) [inline tox.ini]: Sure thing.

**Comment by bz2** [comment]
Okay, 9c07a12 has the nooptionals variant run for 2.7 and 3.6 only for both travis and tox.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**